determined-ai/determined
# feat: add prometheus endpoint for internal Determined state mappings [DET-5890]

## Description



adds an endpoint at /debug/prom/det-metrics that exposes entity mappings needed for DCGM exporter and cAdvisor.

## Test Plan



- enable prometheus config in master.yaml
- create prometheus.yml 
- deploy and hit /debug/prom/det-metrics and verify mappings


## Commentary (optional)



Open questions/concerns:
- do we ship prometheus.yml with this? i didn't include my mapping file because it doesn't seem like it belongs in the default master package that is bundled. however, it's not straightforward to rewrite some of the configs/labels that are necessary for prometheus to appropriately scrape
- only the endpoint checks if the master config has prometheus enabled. all other code that associates mappings will run no matter what. adding an if wrapper around all of these sprinkled associations is very messy, but not doing so is additional overhead. 


## Checklist

See [Release Note](../docs/release-notes/README.md) for details.





[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
